solutions/devsprint-michael-douglas-2/DeliveryApp/Screens/Home/HomeViewController.swift

```diff
@@ -28,17 +28,18 @@ class HomeViewController: UIViewController {
     }
 
     override func viewDidLoad() {
-
+        homeView.updateLoading(with: true)
         navigationController?.navigationBar.prefersLargeTitles = true
-
+        
         deliveryApi.fetchRestaurants { restaurants in
 
             guard let restaurants = restaurants else {
+                self.homeView.updateLoading(with: false)
                 return
             }
```

Comment:
Neste seu cenário, o correto é chamar o `self.homeView.updateLoading(with: false)` logo após o retorno do completion de `fetchRestaurants`. Por que se você deixar dessa forma, o loading só vai sumir quando o `restaurants` retornar `nil`.

```suggestion
        deliveryApi.fetchRestaurants { restaurants in
            self.homeView.updateLoading(with: false)
            
            guard let restaurants = restaurants else {
                return
            }
```

Reply:
Mas acho que precisa ter ali também, como um tratamento de erro do nil né?  eu tinha coloado o loading false no final da chamada do updateView (quando dá o reloadData da TableView), pensei que colocando lá talvez ficaria mais sincrono. mas vou mover de lugar então e colocar no da  fetchRestaurants.

Reply:
Precisa ter ali dentro do `guard`, mas dai vai ser a chamada de outro método que mostraria a `EmptyView` que está sendo desenvolvida.

Reply:
A sua LoadingView aparece antes de chamar o método de `fetchRestaurants` e some assim que o completion retorna.